### PR TITLE
Add menu item to request files for protocols that support upload share links

### LINKED ISSUE #13426: Add menu item to request files for protocols that support upload share links

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

**Comment by AliveDevil** [comment]
Forbidden for Nextcloud.
image

**Comment by dkocher** [comment]
> Forbidden for Nextcloud. 

I cannot reproduce the failure here.

```
POST /ocs/v2.php/apps/files_sharing/api/v1/shares?path=sandbox&shareType=3&publicUpload=true HTTP/1.1
OCS-APIRequest: true
Accept: application/xml
Content-Length: 0
Host: cloud.iterate.ch
Connection: Keep-Alive
User-Agent: Cyberduck/null.null (Mac OS X/12.4) (aarch64)
Cookie: __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; oc7vsmzhpklo=7hqpkfo1vi52pront5v876o5is; oc_sessionPassphrase=H0jhNhmjXh9k5rf0SYLfzyRo6cKa8Gz4tDYp4%2F7S34qC1FO7th%2BPnomCN6fCf07IrOnda54XC7Or05ctJr7zeNIN7LY8aRJZxkp%2FvA4bRSfDjcTFec7AH6byEzMQ6I%2F9
Accept-Encoding: gzip,deflate
Authorization: Basic ****=

HTTP/1.1 200 OK
Server: nginx/1.20.0
…
```

**Comment by dkocher** [comment]
> Forbidden for Nextcloud. 

Added missing error handling in 8905e800.

File: nextcloud/src/test/java/ch/cyberduck/core/nextcloud/NextcloudShareProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Test
+    public void testToUploadUrlNoPassword() throws Exception {
+        final Path home = new DefaultHomeFinderService(session).find();
+        final DescriptiveUrl url = new NextcloudShareProvider(session).toUploadUrl(home, null, new DisabledPasswordCallback());
+        assertNotSame(DescriptiveUrl.EMPTY, url);
+    }
+
+    @Test
+    public void testToDownloadUrlPasswortTooShort() throws Exception {
```
Comment (AliveDevil) [inline nextcloud/src/test/java/ch/cyberduck/core/nextcloud/NextcloudShareProviderTest.java:58]:
```suggestion
    public void testToDownloadUrlPasswordTooShort() throws Exception {
```

File: nextcloud/src/test/java/ch/cyberduck/core/nextcloud/NextcloudShareProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            });
+            fail();
+        }
+        catch(AccessDeniedException e) {
+            assertEquals("Forbidden. Das Passwort muss mindestens 10 Zeichen lang sein. Please contact your web hosting service provider for assistance.", e.getDetail());
+        }
+    }
+
+    @Test
+    public void testToDownloadUrlPasswort() throws Exception {
```
Comment (AliveDevil) [inline nextcloud/src/test/java/ch/cyberduck/core/nextcloud/NextcloudShareProviderTest.java:75]:
```suggestion
    public void testToDownloadUrlPassword() throws Exception {
```

File: nextcloud/src/test/java/ch/cyberduck/core/nextcloud/NextcloudShareProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            @Override
+            public Credentials prompt(final Host bookmark, final String title, final String reason, final LoginOptions options) {
+                return new Credentials(null, new AlphanumericRandomStringService(10).random());
+            }
+        });
+        assertNotSame(DescriptiveUrl.EMPTY, url);
+    }
+
+    @Test
+    public void testToUploadUrlPasswortTooShort() throws Exception {
```
Comment (AliveDevil) [inline nextcloud/src/test/java/ch/cyberduck/core/nextcloud/NextcloudShareProviderTest.java:116]:
```suggestion
    public void testToUploadUrlPasswordTooShort() throws Exception {
```

**Review by AliveDevil — CHANGES_REQUESTED** [review]

**Pull request merged**